### PR TITLE
DDS-1513 Integrate Design Tokens (1/3: Base Integration)

### PR DESCRIPTION
**Pull request by yodas7**
This PR brings the foundation for a new design token system.
To avoid bloating the PR, the actual component application has been split into #105 and #106. Please refer to these PRs for usage instructions.
For the migration, the oldest token usage in some components has been updated to the previous token usage.
## New features
- A Tailwind CSS plugin to integrate design tokens via CSS variables
  - An equivalent plugin is implemented in the dsv-rp/tailwind repo.
- An ESLint rule to ensure that nonexistent design tokens are not used
- TypeScript-ification of ESLint and Tailwind CSS config
- Updated implementation of Storybook's theme switching feature
- Renamed the prefix of `--dds-` custom CSS properties in some components to `--ddc-`
  - This is a change to avoid a conflict with the `--dds-` prefix used by tokens
  - `ddc` stands for Daikin Design system Component. It means that it is a local variable for the component.

Currently, there is code for the Tailwind CSS plugin in the repository, but if a new version of dsv-rp/tailwind is released, we can use that. This will be done in another PR.

Split from #76.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -149,5 +151,8 @@
     "vite-plugin-externalize-deps": "^0.8.0",
     "vitest": "^2.1.6",
     "web-component-analyzer": "^2.0.0"
+  },
+  "peerDependencies": {
+    "@daikin-oss/dds-tokens": ">=0.3.0 <1.0.0"
```
Comment (hiro-daikin) [inline package.json:156]: This is fine - we just need to mention this in our documentation

File: src/index.ts [hunk 2]
```diff
@@ -1,3 +1,2 @@
-export * from "./colors";
```
Comment (hiro-daikin) [inline src/index.ts:1]: Dosatsu currently uses the color palette. 
Where and how would users get the color palette values from?

**Review by hiro-daikin — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (yodas7) [reply src/index.ts:1]: @hiro-daikin 
If users need color codes, they should import them from `@daikin-oss/dds-tokens`.
On the other hand, using color codes directly will result in a lack of theme support, so if possible, I would recommend using them in a way like `var(--dds-color-blue-40)`.

In any case, I think that these are the responsibilities of `@daikin-oss/dds-tokens` and `@daikin-oss/plugin` now that the design tokens have been established.

File: package.json (hunk 1 above)
Comment (yodas7) [reply package.json:156]: I've updated README to install `@daikin-oss/dds-tokens` as well.
https://github.com/dsv-rp/DDS/pull/105/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

File: src/index.ts (hunk 2 above)
Comment (hiro-daikin) [reply src/index.ts:1]: I agree that it should probably be the responsibility of other packages.
This is a breaking change, however, and should be included in the changeset

File: package.json (hunk 1 above)
Comment (hiro-daikin) [reply package.json:156]: @yodas7 Hopefully this is also included in our zeroheight documentation

**Review by hiro-daikin — APPROVED** [review]

**Review by ukkary — APPROVED** [review]

**Pull request merged**